app/backend/src/couchers/servicers/api.py

```diff
@@ -657,6 +671,38 @@ def SendFriendRequest(self, request, context, session):
         ):
             context.abort(grpc.StatusCode.FAILED_PRECONDITION, errors.FRIENDS_ALREADY_OR_PENDING)
 
+        # Check if user has been sending friend requests excessively
+        count_friend_requests_last_24h = session.execute(
+            select(func.count())
+            .select_from(FriendRelationship)
+            .where(FriendRelationship.from_user_id == context.user_id)
+            .where(FriendRelationship.time_sent >= now() - timedelta(hours=24))
+        ).scalar_one()
+        if count_friend_requests_last_24h >= FRIEND_REQUEST_DAILY_BLOCKING_QUOTA - 1:
```

Comment:
Why the `-1`?

Reply:
So it blocks on the 50th request if FRIEND_REQUEST_DAILY_BLOCKING_QUOTA is 50, as the count are just the already made requests.

Reply:
Well if there is a limit of n requests, you would assume the user will be limited on their n+1th req? So that they can do exactly n and no more?

Reply:
Changed it now, yeah I guess that makes sense 👍